src/models/ratings/ratings.service.spec.ts

```diff
@@ -0,0 +1,321 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { Ratings } from './entities/ratings.entity';
+import { RatingService } from './ratings.service';
+import { Players } from './entities/players.entity';
+import { Games } from '../games-history/entities/games.entity';
+import exp from 'constants';
+
+
+describe('RatingService', () => {
+	let service: RatingService;
+
+	const mockRepo = {
+		findAndCount: jest.fn().mockImplementation(() => [[{ sn: '1234' }], 1]),
+		findOne: jest.fn(),
+		findByIds: jest.fn(),
+		find: jest.fn(),
+		save: jest.fn(),
+		update: jest.fn(),
+		delete: jest.fn(),
+		createQueryBuilder: jest.fn(() => ({
+			select: () => jest.fn(),
+			where: () => jest.fn(),
+			orWhere: () => jest.fn(),
+			from: () => jest.fn(),
+			whereInIds: () => jest.fn(),
+			orderBy:() => jest.fn(),
+			groupBy: () => jest.fn(),
+			delete: () => jest.fn(),
+			execute: () => jest.fn(),
+		})),
+		manager: {
+			connection: {
+				transaction: jest.fn(),
+			},
+		},
+	};
+	const INITIAL_RATING = 1000;
+	const BONUS_RATING = 750;
+	const BONUS_FACTOR = 60;
+	const PARTICIPATION_LIMIT = 10;
+	const PARTICIPATION_CUTOFF = 1500;
+	const MAX_DROP = 200;
+	const RATING_RETENTION = 1000 * 60 * 60 * 24 * 240;
+		
+	beforeEach(async () => {
+		const module: TestingModule = await Test.createTestingModule({
+			providers: [
+				RatingService,
+				{
+					provide: getRepositoryToken(Ratings, 'default'),
+					useValue: mockRepo,
+				},
+				{
+					provide: getRepositoryToken(Players, 'default'),
+					useValue: mockRepo,
+				},
+				{
+					provide: getRepositoryToken(Games, 'games'),
+					useValue: mockRepo,
+				},
+			],
+		}).compile();
+
+		service = module.get<RatingService>(RatingService);
+	});
+
+	it('should be defined', () => {
+		expect(service).toBeDefined();
+	});
+	
+	// describe('Generate Ratings', () => {	
```

Comment:
?

Reply:
more so just left this in here to come back and add more unit tests for the generateRating method but I'll just go ahead and remove it, as the main business logic has testing.